marklogic-data-hub-central/ui/src/pages/Browse.tsx

```diff
@@ -148,13 +148,19 @@ const Browse: React.FC<Props> = ({location}) => {
     if (searchOptions.nextEntityType && searchOptions.nextEntityType !== "All Data") {
       setCardView(false);
     }
-    if ((entities.length || (searchOptions.nextEntityType === "All Data" || searchOptions.nextEntityType === "All Entities" || searchOptions.nextEntityType === undefined)) &&
-      (!searchOptions.nextEntityType ||
-        (searchOptions.nextEntityType === "All Entities" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && !cardView) ||
+    if ((entities.length > 0 || (searchOptions.nextEntityType === "All Data" || searchOptions.nextEntityType === "All Entities" || searchOptions.nextEntityType === undefined)) &&
+      ((!searchOptions.nextEntityType && entities.length > 0) ||
+        (searchOptions.nextEntityType === "All Entities" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && !cardView && entities.length > 0) ||
         (searchOptions.nextEntityType === "All Data" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && cardView) ||
-        (!["All Entities", "All Data"].includes(searchOptions.nextEntityType) && searchOptions.entityTypeIds[0] === searchOptions.nextEntityType)
+        (searchOptions.nextEntityType && !["All Entities", "All Data"].includes(searchOptions.nextEntityType) && searchOptions.entityTypeIds[0] === searchOptions.nextEntityType)
       )) {
       getSearchResults(entities);
+    } else {
+      if (!cardView && (!searchOptions.entityTypeIds.length || !searchOptions.nextEntityType)) {
```

Comment:
This condition crashes the app due to setting facet object to be empty if apply facets and refresh the page or open detail page and return back to browse. 
<img width="1088" alt="Screen Shot 2021-02-09 at 8 29 08 AM" src="https://user-images.githubusercontent.com/44847151/107394912-32dc8300-6ab1-11eb-8ed2-8365c5937e66.png">

Reply:
This is resolved.